diff --git a/client/src/OAuth2Demo/Client/Controllers/CountEggs.php b/client/src/OAuth2Demo/Client/Controllers/CountEggs.php
index 94bdfb0..418c8c7 100644
--- a/client/src/OAuth2Demo/Client/Controllers/CountEggs.php
+++ b/client/src/OAuth2Demo/Client/Controllers/CountEggs.php
@@ -40,6 +40,11 @@ class CountEggs extends BaseController
         $request = $http->post('/api/'.$user->coopUserId.'/eggs-count');
         $request->addHeader('Authorization', 'Bearer '.$user->coopAccessToken);
         $response = $request->send();
+
+        if ($response->isError()) {
+            throw new \Exception($response->getBody(true));
+        }
+
         $countEggsData = json_decode($response->getBody(), true);
 
         $eggCount = $countEggsData['data'];
