diff --git a/client/src/OAuth2Demo/Client/Controllers/CountEggs.php b/client/src/OAuth2Demo/Client/Controllers/CountEggs.php
index 71e16fc..93c5058 100644
--- a/client/src/OAuth2Demo/Client/Controllers/CountEggs.php
+++ b/client/src/OAuth2Demo/Client/Controllers/CountEggs.php
@@ -32,10 +32,10 @@ class CountEggs extends BaseController
         $request = $http->post('/api/'.$user->coopUserId.'/eggs-count');
         $request->addHeader('Authorization', 'Bearer '.$user->coopAccessToken);
         $response = $request->send();
-        echo ($response->getBody(true));die;
         $countEggsData = json_decode($response->getBody(), true);
 
-        die('Implement this in CountEggs::countEggs');
+        $eggCount = $countEggsData['data'];
+        $this->setTodaysEggCountForUser($this->getLoggedInUser(), $eggCount);
 
         return $this->redirect($this->generateUrl('home'));
     }
