diff --git a/client/src/OAuth2Demo/Client/Controllers/CountEggs.php b/client/src/OAuth2Demo/Client/Controllers/CountEggs.php
index e1fdb7a..d187351 100644
--- a/client/src/OAuth2Demo/Client/Controllers/CountEggs.php
+++ b/client/src/OAuth2Demo/Client/Controllers/CountEggs.php
@@ -2,6 +2,8 @@
 
 namespace OAuth2Demo\Client\Controllers;
 
+use Guzzle\Http\Client;
+
 class CountEggs extends BaseController
 {
     public static function addRoutes($routing)
@@ -19,6 +21,17 @@ class CountEggs extends BaseController
      */
     public function countEggs()
     {
+        $http = new Client('http://coop.apps.knpuniversity.com', array(
+            'request.options' => array(
+                'exceptions' => false,
+            )
+        ));
+
+        $request = $http->post('/api/me');
+        $request->addHeader('Authorization', 'Bearer '.$accessToken);
+        $response = $request->send();
+        $json = json_decode($response->getBody(), true);
+
         die('Implement this in CountEggs::countEggs');
 
         return $this->redirect($this->generateUrl('home'));
