diff --git a/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php b/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
index 64b5e2f..825dbf1 100644
--- a/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
+++ b/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
@@ -24,7 +24,14 @@ class CoopOAuthController extends BaseController
      */
     public function redirectToAuthorization(Request $request)
     {
-        die('Hallo world!');
+        $url = 'http://coop.apps.knpuniversity.com/authorize?'.http_build_query(array(
+            'response_type' => 'code',
+            'client_id' => '?',
+            'redirect_uri' => '?',
+            'scope' => 'eggs-count profile'
+        ));
+
+        var_dump($url);die;
     }
 
     /**
