diff --git a/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php b/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
index a28f62c..305ca85 100644
--- a/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
+++ b/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
@@ -71,6 +71,7 @@ class CoopOAuthController extends BaseController
         $responseArr = json_decode($responseBody, true);
         $accessToken = $responseArr['access_token'];
         $expiresIn = $responseArr['expires_in'];
+        $expiresAt = new \DateTime('+'.$expiresIn.' seconds');
 
         $request = $http->get('/api/me');
         $request->addHeader('Authorization', 'Bearer '.$accessToken);
@@ -80,6 +81,7 @@ class CoopOAuthController extends BaseController
         $user = $this->getLoggedInUser();
         $user->coopAccessToken = $accessToken;
         $user->coopUserId = $json['id'];
+        $user->coopAccessExpiresAt = $expiresAt;
         $this->saveUser($user);
 
         die('Implement this in CoopOAuthController::receiveAuthorizationCode');
