diff --git a/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php b/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
index 8c9800b..b045c47 100644
--- a/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
+++ b/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
@@ -98,6 +98,7 @@ class CoopOAuthController extends BaseController
         $user->coopAccessExpiresAt = $expiresAt;
         $this->saveUser($user);
 
-        die('Implement this in CoopOAuthController::receiveAuthorizationCode');
+        // redirect back to the homepage
+        return $this->redirect($this->generateUrl('home'));
     }
 }
