diff --git a/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php b/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php
index 667d8e8..680922b 100644
--- a/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php
+++ b/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php
@@ -51,6 +51,13 @@ class FacebookOAuthController extends BaseController
         $facebook = $this->createFacebook();
 
         $userId = $facebook->getUser();
+
+        if (!$userId) {
+            return $this->render('failed_authorization.twig', array(
+                'response' => $request->query->all()
+            ));
+        }
+
         var_dump($userId);die;
 
         die('Todo: Handle after Facebook redirects to us');
