diff --git a/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php b/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php
index d9781fa..d6bd69f 100644
--- a/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php
+++ b/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php
@@ -93,7 +93,7 @@ class FacebookOAuthController extends BaseController
         return $this->redirect($this->generateUrl('home'));
     }
 
-    private function makeApiRequest(\Facebook $facebook, $url, $method, $parameters)
+    private function makeApiRequest(\Facebook $facebook, $url, $method, $parameters, $retry = true)
     {
         try {
             return $facebook->api($url, $method, $parameters);
@@ -104,6 +104,11 @@ class FacebookOAuthController extends BaseController
                 return $this->redirect($this->generateUrl('facebook_authorize_start'));
             }
 
+            // re-try one time
+            if ($retry) {
+                return $this->makeApiRequest($facebook, $url, $method, false);
+            }
+
             // it failed for some odd reason...
             throw $e;
         }
