diff --git a/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php b/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php
index 7610d8c..843f90c 100644
--- a/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php
+++ b/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php
@@ -58,6 +58,13 @@ class FacebookOAuthController extends BaseController
             ));
         }
 
+        try {
+            $json = $facebook->api('/me');
+        } catch (\FacebookApiException $e) {
+            return $this->render('failed_token_request.twig', array('response' => $e->getMessage()));
+        }
+        var_dump($json);die;
+
         if ($this->isUserLoggedIn()) {
             $user = $this->getLoggedInUser();
         } else {
