diff --git a/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php b/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php
index cf2ddbf..8703825 100644
--- a/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php
+++ b/client/src/OAuth2Demo/Client/Controllers/FacebookOAuthController.php
@@ -25,13 +25,20 @@ class FacebookOAuthController extends BaseController
     public function redirectToAuthorization()
     {
         $config = array(
-          'appId' => 'YOUR_APP_ID',
-          'secret' => 'YOUR_APP_SECRET',
+          'appId' => '1386038978283395',
+          'secret' => '9ec32a48f1ad1988e0d4b9e80a17d5bc',
           'allowSignedRequest' => false // optional but should be set to false for non-canvas apps
         );
 
         $facebook = new \Facebook($config);
 
+        $redirectUrl = $this->generateUrl('facebook_authorize_redirect', array(), true);
+
+        $url = $facebook->getLoginUrl(array(
+            'redirect_uri' => $redirectUrl,
+            'scope' => array('publish_actions', 'email')
+        ));
+
         die('Todo: Redirect to Facebook');
     }
 
