diff --git a/client/views/dashboard.twig b/client/views/dashboard.twig
index 3bf55c3..cc0cbb6 100644
--- a/client/views/dashboard.twig
+++ b/client/views/dashboard.twig
@@ -49,7 +49,14 @@
                 // prevent the click from going to #
                 e.preventDefault();
 
-                gapi.auth.signIn();
+                var myParams = {
+                    'clientid': '104029852624-a72k7hnbrrqo02j5ofre9tel76ui172i.apps.googleusercontent.com',
+                    'cookiepolicy': 'single_host_origin',
+                    'callback': 'mySignInCallback',
+                    'scope': 'https://www.googleapis.com/auth/plus.login',
+                    'requestvisibleactions': 'http://schemas.google.com/AddActivity'
+                };
+                gapi.auth.signIn(myParams);
             });
         });
     </script>
