diff --git a/client/views/dashboard.twig b/client/views/dashboard.twig
index 8f62ff1..71e162a 100644
--- a/client/views/dashboard.twig
+++ b/client/views/dashboard.twig
@@ -61,8 +61,14 @@
             });
         });
 
+        var isSignedIn = false;
         function mySignInCallback(authResult) {
             if (authResult['status']['signed_in']) {
+                if (isSignedIn) {
+                    return;
+                }
+                isSignedIn = true;
+
                 // Update the app to reflect a signed in user
                 // Hide the sign-in button now that the user is authorized, for example:
                 $('.js-google-signin').hide();
