diff --git a/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php b/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
index 27112fa..ba27f99 100644
--- a/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
+++ b/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
@@ -108,6 +108,8 @@ class CoopOAuthController extends BaseController
                 $json['firstName'],
                 $json['lastName']
             );
+
+            $this->loginUser($user);
         }
         $user->coopAccessToken = $accessToken;
         $user->coopUserId = $json['id'];
