diff --git a/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php b/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
index 49743ad..8c37c8b 100644
--- a/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
+++ b/client/src/OAuth2Demo/Client/Controllers/CoopOAuthController.php
@@ -81,6 +81,7 @@ class CoopOAuthController extends BaseController
         $response = $request->send();
         $responseBody = $response->getBody(true);
         $responseArr = json_decode($responseBody, true);
+        var_dump($responseArr);die;
 
         // if there is no access_token, we have a problem!!!
         if (!isset($responseArr['access_token'])) {
