diff --git a/client/data/refresh_tokens.php b/client/data/refresh_tokens.php
index 670651e..ad1d5d0 100644
--- a/client/data/refresh_tokens.php
+++ b/client/data/refresh_tokens.php
@@ -21,9 +21,8 @@ foreach ($expiringTokens as $userInfo) {
     $request = $http->post('/token', null, array(
         'client_id'     => 'TopCluck',
         'client_secret' => '2e2dfd645da38940b1ff694733cc6be6',
-        'grant_type'    => 'authorization_code',
-        'code'          => $code,
-        'redirect_uri'  => $this->generateUrl('coop_authorize_redirect', array(), true),
+        'grant_type'    => 'refresh_token',
+        'refresh_token' => $userInfo['coopRefreshToken'],
     ));
 
     // make a request to the token url
