diff --git a/cron/collect_eggs.php b/cron/collect_eggs.php
index 80023a3..cc3e981 100644
--- a/cron/collect_eggs.php
+++ b/cron/collect_eggs.php
@@ -20,9 +20,8 @@ $request = $http->post('/token', null, array(
 // make a request to the token url
 $response = $request->send();
 $responseBody = $response->getBody(true);
-var_dump($responseBody);die;
-
-$accessToken = 'GET THIS FROM YOUR APPLICATION FOR NOW';
+$responseArr = json_decode($responseBody, true);
+$accessToken = $responseArr['access_token'];
 
 $request = $http->post('/api/2/eggs-collect');
 $request->addHeader('Authorization', 'Bearer '.$accessToken);
